spec/io/udp_sockets_spec.rb

```diff
@@ -95,5 +95,45 @@ module Cosmos
     end
 
   end
+
+  describe UdpReadWriteSocket do
+
+    describe "initialize" do
+      it "creates a socket" do
+        udp = UdpReadWriteSocket.new(8888)
+        expect(udp.local_address.ip_address).to eql '0.0.0.0'
+        expect(udp.local_address.ip_port).to eql 8888
+        udp.close
+        if RUBY_ENGINE == 'ruby' # UDP multicast does not work in Jruby
```

Comment:
UDP multicast doesn't work in JRuby? Why? How do we document that?

Reply:
I think it was because of this:

        if (address == null) {
            if ( non_block ) return null; // :wait_readable or raise WaitReadable
            throw runtime.newErrnoEAGAINReadableError("multicast UDP does not support nonblocking");
        }

But I don't remember.  We could probably make it work with some compromise (use blocking sockets), but I didn't deal with it in the initial port to jruby.   I'll write a ticket.